### ISSUE TITLE
comparing to #embed

### ISSUE THREAD

**Issue by johannes-graeter** [MKlimenko/embed#3]
Hi, 

thanks for the great work!
C23 added #embed https://en.cppreference.com/w/c/preprocessor/embed do you some benchmark that compares this to your lib? Any insight on when to use yours or #embed?

Best, Johannes

**Comment by MKlimenko** [comment]
Hi Johannes,

Personally, I'd use the standard way, that way you'll get rid of an extra step in the build procedure.
However, looks like no compiler supports `#embed` so far, but I haven't been following that lately. Some sort of custom trunk was once live on Compiler Explorer, but it might be down now.

My library was inspired by `std::embed` proposal of the same author, but (long story short) this functionality landed in plain C first. Therefore, I designed it to be a resourse- rather than an array of bytes-oriented to be more C++ish.

It worked pretty well on my previous job for storing the binary images of the heterogeneous device programs, we integrated it into CMake as a pre-build step, so as a conclusion I'd say that it's pretty straightforward tool to use until we have a standard way.
